webapp/src/pages/GameRulesPage/GameRulesPage.tsx

```diff
@@ -0,0 +1,148 @@
+import { gameQueueClientToServerWsEvents } from 'transcendence-shared';
+import {
+  ButtonProps,
+  ButtonVariant,
+  Header,
+  IconVariant,
+  ReactiveButton,
+  Text,
+  TextVariant,
+} from '../../shared/components';
+import { GamePairingStatusDtoGameQueueStatusEnum } from '../../shared/generated';
+import { useGamePairing } from '../../shared/hooks/UseGamePairing';
+import { useNavigation } from '../../shared/hooks/UseNavigation';
+import socket from '../../shared/socket';
+import { PLAY_GAME_TRAIN_URL } from '../../shared/urls';
+import './GameRulesPage.css';
+
+export default function GameRulesPage() {
+  const { goBack, navigate } = useNavigation();
+  const { gameQueueStatus } = useGamePairing();
+
+  let buttons = [
+    {
+      variant: ButtonVariant.SUBMIT,
+      iconVariant: IconVariant.PLAY,
+      onClick: () => navigate(PLAY_GAME_TRAIN_URL),
+      children: 'Training',
+    },
+  ];
+
+  if (gameQueueStatus === GamePairingStatusDtoGameQueueStatusEnum.None) {
+    buttons.push({
+      variant: ButtonVariant.SUBMIT,
+      iconVariant: IconVariant.ADD,
+      onClick: () => {
+        socket.emit(gameQueueClientToServerWsEvents.gameQueueJoin);
+      },
+      children: 'Join game queue',
+    });
+  }
+
```

Comment:
I like how you did this, one or two buttons depending on the gameQueueStatus

Reply:
Thanks!